spring-web/src/main/java/org/springframework/web/server/ServerWebExchangeDecorator.java

```diff
@@ -82,6 +82,14 @@ public ServerHttpResponse getResponse() {
 		return getDelegate().getSession();
 	}
 
+	@Override
+	public Mono<WebSession> getSessionOrEmpty() {
+		if (getDelegate().getSession() == null) {
+			return Mono.empty();
+		}
+		return getSession();
+	}
```

Comment:
All three implementations are the same. How about providing a default implementation in `ServerWebExchange`?

Reply:
All right with that. getSessionOrEmpty() using already created method getSession(). Difference between getSessionOrEmpty() and getSession() in checking for null

Reply:
I mean like below.
```
	default Mono<WebSession> getSessionOrEmpty() {
		if (getSession() == null) {
			return Mono.empty();
		}
		return getSession();
	}
```